integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/SparkPropertyFacetBuilder.java

```diff
@@ -8,26 +8,42 @@
 import io.openlineage.spark.agent.facets.SparkPropertyFacet;
 import io.openlineage.spark.api.CustomFacetBuilder;
 import io.openlineage.spark.api.OpenLineageContext;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 import java.util.function.BiConsumer;
 import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.spark.SparkConf;
+import org.apache.spark.SparkContext;
+import org.apache.spark.scheduler.SparkListenerEvent;
 import org.apache.spark.scheduler.SparkListenerJobStart;
+import org.apache.spark.sql.SparkSession;
 
+@Slf4j
 public class SparkPropertyFacetBuilder
-    extends CustomFacetBuilder<SparkListenerJobStart, SparkPropertyFacet> {
+    extends CustomFacetBuilder<SparkListenerEvent, SparkPropertyFacet> {
```

Comment:
What's the reason in changing this and adding extra check for `SparkListenerJobStart` in buildFacet method? This will result in adding facet for all the events while keeping it empty most of the time.

Reply:
@pawel-big-lebowski 
Sorry for replying your comments too late. I will answer your questions here.
We want to make sure SparkRunFacet be recorded not only at JobStart, but also at JobEnd. This is because attributes in run-time config might change during the running of a job. However, Java doen't support grammar like 'public void build(SparkListenerJobStart|SparkListenerJobEnd event)'.  So we change the parameter of this function from SparkListenerJobStart to SparkListenerEvent.

Reply:
Ok, but this will also result adding facet to sql start, sql complete, and other possible events. If you want to add it ONLY for SparkListenerJobStart and SparListnerJobEnd and it's necessary to add condition in `io.openlineage.spark.agent.facets.builder.SparkPropertyFacetBuilder#build` which does not run
`consumer.accept("spark_properties", new SparkPropertyFacet(m));`
for other listener events that `SparkListenerJobStart` and `SparListnerJobEnd`. In other words, `build` method can start with:

```
if  (!(event instanceOf SparkListenerJobStart) && !(event instanceOf SparkListenerJobEnd) {
   return;
}
```

Reply:
@pawel-big-lebowski  
Got that. I have changed my code as you suggested.